### PR TITLE
Test every weapon and vehicle call

### PR DESCRIPTION
**Pull request by ejdaduya**

### CODE REVIEW

**Review by Girbons — CHANGES_REQUESTED** [review]
Hi @ejdaduya I really appreciate your pull request but as i suggested [here](https://github.com/Girbons/battlefield1-sdk/issues/11) i'd like to have explicit tests.
Could you please follow the suggested code?
Thanks very much for your help.

**Comment by ejdaduya** [comment]
Hi, as stated in line [86](https://github.com/Girbons/battlefield1-sdk/pull/12/files#diff-30073e0d5d029e66ae7bfea448e38a19R86) & line [102](https://github.com/Girbons/battlefield1-sdk/pull/12/files#diff-30073e0d5d029e66ae7bfea448e38a19R102), would you like to maybe add them to a log file or execute a specific task?

I've included a parameter as well in case you'd explicitly like to invoke the function.

I could make the explicit tests for you, but I thought you'd appreciate a dynamic function.

I'm pretty new here so please bear with me.

**Comment by Girbons** [comment]
@ejdaduya The submitted code is good but we have to make some changes, remove [this](https://github.com/ejdaduya/battlefield1-sdk/blob/596aa9bcf59188d956ff03294b2e9b93b55237f4/tests/test_battlefield.py#L11)
 and the vehicles and weapons list

In the dynamic function I thought to do something like this:
Using the resolve weapon that you can find [here](https://github.com/Girbons/battlefield1-sdk/blob/master/bf1/resolvers.py#L42)

```python
def weapon_call(weapon):
    bf = Battlefield('girbons', API_KEY, 'Pc')
    response = bf.progression_service.get_weapon(weapon=resolve_weapon(weapon))
    return response
```
Now you can use the dynamic function

```python
def test_selbstlader_optical():
    response = weapon_call('selbstlader m1916 optical')
    assert response.status_code == 200
```

We can isolate tests for vehicles and weapons in 2 files

If you have another idea i'm glad to discuss it :)

**Comment by ejdaduya** [comment]
Should we discard the `test_resolver.py` and create a `test_weapon_resolver.py` and `test_vehicle_resolver.py` then?

**Comment by Girbons** [comment]
@ejdaduya yes

File: tests/test_weapon_resolver.py [hunk 1]
```diff
@@ -0,0 +1,882 @@
...
+    response = weapon_call('selbstlader 1906 factory')
+    assert response.status_code == 200
+
+
+def test_12g_automatic_backbored():
+    response = weapon_call('12g automatic backbored')
+    assert response.status_code == 200
+
+
+def test_lawrence_of_arabia's_SMLE():
```
Comment (Girbons) [inline tests/test_weapon_resolver.py]: change it in lower case `test_lawrence_of_arabias_smle`

**Review by Girbons — CHANGES_REQUESTED** [review]
@ejdaduya nice work,  change the last thing and add your name to the [contributors](https://github.com/Girbons/battlefield1-sdk/blob/master/contributors.rst) file and i will be happy to merge your pull request

**Comment by ejdaduya** [comment]
Hi @Girbons, I've dropped some of the special characters for the function names. Kindly confirm if these meets your expectation.

**Comment by ejdaduya** [comment]
Hi @Girbons, that's probably all from me for today.

**Comment by Girbons** [comment]
@ejdaduya I will take a look in this afternoon it seems good,
Could you please check with an api_key if the tests are passing?
I don't know why the CI is not using the API_KEY and now I can't check the issue on travis

**Comment by ejdaduya** [comment]
Sure thing. I'll give you a feedback once I have some free time tonight.

**Comment by ejdaduya** [comment]
Hi @Girbons, I've tried running the tests and the `resolver` function does not seem to be applicable for the test functions. 

Here are my current issues:
- `resolve_weapon` and `resolve_vehicle` returns the key value from the config file.
- `bf.progression_service.get_vehicle` and `bf.progression_service.get_weapon` returns a dictionary object, not a requests response, thus `assert response.status_code == 200` would fail as well.

I've tested this by running `pytest tests/test_vehicle_resolver.py` and `pytest tests/test_weapon_resolver.py` and simply providing my `API_KEY` from the `utils` file.

**Comment by Girbons** [comment]
@ejdaduya I missed a step, my bad

**Comment by Girbons** [comment]
Thank you very much @ejdaduya :)

**Comment by ejdaduya** [comment]
Cheers!

**Pull request merged**